### DISCUSSION TITLE
コードブロックの読み上げキャンセルについて

### DISCUSSION THREAD

**Discussion by uinf6** [ciffelia/koe#104] [Q&A] [answered]
DockerもRustも、何もかも初めて触った初心者です。
初心者でもわかりやすいご説明をいただきとても感謝しております。毎日使わせていただいてます☺

この度は、スポイラー（伏字・ネタバレ）の読み上げキャンセルがどこのフォルダでプログラムされているかをお聞きしたくご連絡差し上げました。
スポイラーの読み上げキャンセルのプログラムを参考に、コードブロックの読み上げキャンセルも追加しようとしましたが、スポイラーのプログラムが探せど探せど見つからず…
大変恐縮ではございますがご返信お待ちしております。よろしくお願いいたします。

**Comment by ciffelia** [comment, marked as answer]
こちらです。

https://github.com/ciffelia/koe/blob/main/crates/koe/src/message/read.rs#L23-L26

Markdownの装飾を削除するのにdiscord-mdというクレートを使用しています。コードブロックの変換部分はここにあります。

https://github.com/ciffelia/discord-md/blob/main/src/generate.rs#L214-L225

**Reply by uinf6** [reply]
早急なご返信をいただきありがとうございます！
なるほどクレートと呼ばれるものを使ってスポイラーの読み上げキャンセルを行っていたんですね。
同じようにコードブロック（OneLineCodeとMultiLineCode）をkoe/crates/koe/src/message/read.rsのomit_soiler: true,の後に追加しようとしましたが、コードブロックはスポイラーのようなフィールド（omit_spoiler）がToMarkdownStringOption内に存在せず、ビルド時にエラーが出てしまいました。

私のやり方が間違っているのだとは思いますが、自力ではエラーが解決できず…
新しいフィールドを構築するような技術は私には難しく、今回の件は諦めてもいいかなぁと思っております。
ご教示いただいたにもかかわらず、自己解決できず申し訳ありません😔

もし、koe-1.1.0のバージョンアップをお考えでしたら、コードブロックの読み上げキャンセルも入れていただけますととても嬉しいです。今後ともよろしくお願いいたします☺

**Reply by ciffelia** [reply]
Koe v1.1.1をカスタマイズすることでコードブロックの読み上げキャンセルができるようになりました。

以下のコードを次のように書き換えると読み上げがスキップされるはずです。
https://github.com/ciffelia/koe/blob/1.1.1/crates/koe/src/message/read.rs#L24-L26
```rust
&ToMarkdownStringOption::new()
    .omit_format(true)
    .omit_spoiler(true)
    .omit_one_line_code(true)
    .omit_multi_line_code(true),
```

**Reply by uinf6** [reply]
ciffelia 様
Koeのバージョンアップとカスタマイズのご案内をいただきありがとうございます。

無事、コードブロックの読み上げキャンセルができるようになりました！
より簡潔に読み上げ無しで長文を送れるようになり、とてもうれしく思っております。

これからも重宝いたします。改めてありがとうございます！☺
